tools/preprocess_lancet_intervals.cwl

```diff
@@ -40,22 +44,23 @@ arguments:
               cmd += "vcf2bed --snvs < " + inputs.mutect2_vcf.basename +  " | cut -f 1-3 > mutect2.snvs.bed;";
               bed.push("mutect2.snvs.bed");
           }
-        if(flag == 0){
-            cmd += "echo \"No input vcfs found to convert.  Returning ref bed\"; cp " + inputs.ref_bed.path + " " + inputs.output_basename + ".lancet_intvervals.bed;";
-        }
-        else{
-            cmd += "cat " + bed.join(" ") + " " + inputs.ref_bed.path + " | bedtools sort | bedtools merge > " + inputs.output_basename + ".lancet_intvervals.bed;";
+          if(flag == 0){
+              cmd += "echo \"No input vcfs found to convert.  Returning ref bed\"; cp " + inputs.ref_bed.path + " " + inputs.output_basename + ".lancet_intvervals.bed;";
+          }
+          else{
+              cmd += "cat " + bed.join(" ") + " " + inputs.ref_bed.path + " | bedtools sort | bedtools merge > " + inputs.output_basename + ".lancet_intvervals.bed;";
+          }
+          return cmd;
         }
-        return cmd;
       }
 
 inputs:
   strelka2_vcf: {type: ['null', File], doc: "PASS vcf from strelka2 run for the sample to be analyzed"}
   mutect2_vcf: {type: ['null', File], doc: "PASS vcf from mutect2 run for the sample to be analyzed"}
-  ref_bed: {type: File, doc: "Exonic bed file recommended.  Will be augmented with strelka2 and/or mutect2 sites if supplied"}
+  ref_bed: {type: 'File?', doc: "Optional exonic bed file recommended.  Will be augmented with strelka2 and/or mutect2 sites if supplied"}
```

Comment:
I think my explanation was too vague, you should either supply a WGS intervals file, but not typically recommended for lancet given its slow run time; the preferred way is to use intervals from essentially a gtf, then this tool will supplement the search space with strelka2 and mutect2 calls for validation, as the NYGC, creators of this tool did, for a paper. Makes sense?

Reply:
so what im doing is kind of passing the failure buck to get around the lancet preprocessing stuff only happening in WGS. the inital gatk_intervallisttool step should always run and generate an output. however the second gatk_intervallisttool step (exome_plus) only runs after that preprocess lancet interval step step in WGS. the preprocess step takes a bed file that is only provided in WGS. so whats the way around it? if the bed file isnt provided, then preprocess_lancet_intervals isnt run and passes a null file, that null file ends up at gatk_intervallisttool_exome_plus and triggers it to not run and also pass an empty file, which then gets to a mode_selector step called select_lancet_bed_inteval. only there is the failure caught. if the mode is WGS and we dont have the output from gatk_intervallisttool_exome_plus it will fail. if the mode is WXS and we dont have the intial gatk_intervallisttool output, it will fail.

Reply:
hmmm, ok, i think i see what you mean